src/signatures/schnorr.rs

```diff
@@ -70,6 +107,63 @@ where
         Ok(Self::new(public_nonce, s))
     }
 
+    /// Signs a message with the given secret key.
+    ///
+    /// This method correctly binds a nonce and the public key to the signature challenge, using domain-separated
+    /// hashing. The hasher is also opinionated in the sense that Blake2b 256-bit digest is always used.
+    ///
+    /// it is possible to customise the challenge by using [`construct_domain_separated_challenge`] and [`sign_raw`]
+    /// yourself, or even use [`sign_raw`] using a completely custom challenge.
+    pub fn sign_message<'a, B>(secret: &'a K, message: B) -> Result<Self, SchnorrSignatureError>
+    where
+        K: Add<Output = K> + Mul<&'a K, Output = K>,
+        B: AsRef<[u8]>,
+    {
+        let nonce = K::random(&mut rand::thread_rng());
+        let public_nonce = P::from_secret_key(&nonce);
+        let public_key = P::from_secret_key(secret);
+        let challenge = Self::construct_domain_separated_challenge::<_, Blake256>(&public_nonce, &public_key, message);
```

Comment:
See other comment about swapping the first two challenge inputs.

Reply:
We've pretty much standardised on this order. Swapping them at this stage produces greater risk of bugs.